daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -433,7 +433,12 @@ private[lf] object SBuiltin {
   final case object SBCodePointsToText extends SBuiltinPure(1) {
     override private[speedy] def executePure(args: util.ArrayList[SValue]): SText = {
       val codePoints = getSList(args, 0).map(_.asInstanceOf[SInt64].value)
-      SText(Utf8.pack(codePoints.toImmArray))
+      Utf8.pack(codePoints.toImmArray) match {
+        case Right(value) =>
+          SText(value)
+        case Left(cp) =>
+          crash(s"invalid code point 0x${cp.toHexString}.")
```

Comment:
Maybe  this one should be an `ArithmeticError` :cry:

Reply:
How hard would it be to make this dependent on the LF version? It seems like we could just have a flag in this primitive on whether it can throw or not and then the speedy compiler can set it to true f we’re in LF >= 1.15 while leaving it false for older versions.

Reply:
I think the simpler would be two have two different builtins in the scala AST. 
If we do not want to have two builtins in the compiler/archive, we can hack the archive decoder to map `CODE_POINTS_TO_TEXT` to one or the other depending of the LF version.

However I am not sure, the trouble is worthwhile. I think I will let the builtin crashing forever.

Reply:
yeah I think we can wait until we get a feature request for it. I’m not all that concerned about it.